### ISSUE TITLE
Airtap 4.0.0

### ISSUE THREAD

**Issue by vweevers** [airtap/airtap#292]
Tracking issue for remaining tasks after the major refactoring of #275. Everything is code complete, I just need to push modularized code to new repositories and setup CI where applicable.

- [x] Push and publish `airtap-default`
- [x] Push and publish `airtap-sauce`  
- [x] Push and publish `airtap-playwright`
- [x] Push and publish `airtap-electron`
- [x] Push and publish `airtap-system`
- [x] Push and publish `airtap-manual`
- [x] Create `demo` repository (previously `example/` in airtap)

**Comment by vweevers** [comment]
Published [`airtap-playwright`](https://github.com/airtap/playwright). Running playwright in Travis CI [requires ubuntu 18.04 and a few apt packages](https://github.com/airtap/playwright/blob/master/.travis.yml).

If you want to try it out: `npm i airtap/airtap airtap-playwright`.

**Comment by vweevers** [comment]
Published [`airtap-electron`](https://github.com/airtap/electron).

**Comment by vweevers** [comment]
Published [`airtap-system`](https://github.com/airtap/system). Not stable yet. Works well on windows, stalls on linux, haven't tried on mac yet.

**Comment by tinchoz49** [comment]
Hey @vweevers, this is really cool! I was testing here https://github.com/geut/discovery-swarm-webrtc and it works great!!

One thing that I want to ask you: How can I define the `wants` prop in the manifest ?

I try to put this on `.airtap.yml`:

```yml
providers:
  - airtap-playwright

server: './bin/index.js'

browsers:
  - name: firefox
    wants:
      loopback: '...'
```

but I got: `No manifest found matching`

**Comment by vweevers** [comment]
> How can I define the wants prop in the manifest ?

Only providers can define properties, which airtap then matches against what you specify in `.airtap.yml`. Can you explain why you want to set [`wants.loopback`](https://github.com/airtap/browser-manifest#wants)? There might be another way.

**Comment by tinchoz49** [comment]
Ah ok. Well, I'm testing simple-peer connections and in firefox if the hostname is "localhost" it throws an error but if I use 127.0.0.1 it works. I was trying to find a way to customize the url that the provider access to.

**Comment by vweevers** [comment]
Interesting. Who is throwing an error? Could you provide a simple repro with instructions on how to install and run? Currently there is no way to customize the url; the loopback logic is only intended for the Sauce Connect tunnel used by `airtap-sauce`.

**Comment by tinchoz49** [comment]
The internal firefox WebRTC throws the error, is not related with airtap (test runner works great). 

I made a repository with to test it: https://github.com/tinchoz49/test-airtap-webrtc

You can:
1. run the dev server
2. start firefox
3. test the p2p connection with `127.0.0.1:9966` and will work,
4. test with `localhost:9966` and it will fail

Weird issue in firefox only, chromium all good.

**Comment by vweevers** [comment]
Can reproduce on both Playwright FF and local FF, don't understand it. To fix, I'll add support for `airtap --loopback 127.0.0.1 test.js` which would make airtap open `http://127.0.0.1:1234` instead of `http://localhost:1234`. Would apply to all browsers, but is an easier fix. 

Doesn't break the tunneling of `airtap-sauce`, the logic being:

- if you set a custom `--loopback <hostname>` we'll use that on any browser, or
- if the browser doesn't support tunneling through `localhost` (as indicated by `wants.loopback === true`) then we'll use `airtap.local`
- else `localhost`

**Comment by tinchoz49** [comment]
> Can reproduce on both Playwright FF and local FF, don't understand it

Me neither :confused: 

I agree with that fix, thanks.

**Comment by vweevers** [comment]
@tinchoz49 Done in 90775510e99e0452c00450a892644d17c094b0ee

**Comment by tinchoz49** [comment]
it works: https://github.com/tinchoz49/test-airtap-webrtc

**Comment by vweevers** [comment]
Published v4.0.0.

![done-downsized](https://user-images.githubusercontent.com/3055345/93663304-38d0b980-fa67-11ea-80db-f3963f0c7a77.gif)
